lang/java/reef-runtime-standalone/src/main/java/org/apache/reef/runtime/standalone/client/StandaloneDriverConfigurationProviderImpl.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.runtime.standalone.client;
+
+import org.apache.reef.runtime.common.client.DriverConfigurationProvider;
+import org.apache.reef.runtime.common.parameters.JVMHeapSlack;
+import org.apache.reef.runtime.standalone.client.parameters.NodeFolder;
+import org.apache.reef.runtime.standalone.client.parameters.NodeListFilePath;
+import org.apache.reef.runtime.standalone.client.parameters.SshPortNum;
+import org.apache.reef.runtime.standalone.driver.StandaloneDriverConfiguration;
+import org.apache.reef.tang.Configuration;
+import org.apache.reef.tang.Configurations;
+import org.apache.reef.tang.annotations.Parameter;
+import org.apache.reef.tang.formats.ConfigurationModule;
+
+import javax.inject.Inject;
+import java.io.*;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Helper class that assembles the driver configuration when run on the local runtime.
+ */
+final class StandaloneDriverConfigurationProviderImpl implements DriverConfigurationProvider {
+
+  private static final Logger LOG = Logger.getLogger(StandaloneDriverConfigurationProviderImpl.class.getName());
+  private final double jvmHeapSlack;
+  private final String nodeListFilePath;
+  private final String nodeFolder;
+  private final int sshPortNum;
+  private final Set<String> nodeInfoSet;
+
+  @Inject
+  StandaloneDriverConfigurationProviderImpl(@Parameter(JVMHeapSlack.class) final double jvmHeapSlack,
+                                            @Parameter(NodeListFilePath.class) final String nodeListFilePath,
+                                            @Parameter(NodeFolder.class) final String nodeFolder,
+                                            @Parameter(SshPortNum.class) final int sshPortNum) {
+    this.jvmHeapSlack = jvmHeapSlack;
+    this.nodeListFilePath = nodeListFilePath;
+    this.nodeFolder = nodeFolder;
+    this.sshPortNum = sshPortNum;
+    this.nodeInfoSet = new HashSet<>();
+
+    LOG.log(Level.FINEST, "Reading NodeListFilePath");
+    try {
```

Comment:
This whole code could be simplified greatly via [Apache Commons IO](https://commons.apache.org/proper/commons-io/description.html)